File: app/src/main/java/com/example/unsplash/model/datasource/PhotoDataSourceFactory.java
package com.example.unsplash.model.datasource;

import android.arch.lifecycle.MutableLiveData;
import android.arch.paging.DataSource;
import android.arch.paging.PageKeyedDataSource;

import com.example.unsplash.model.models.Photo;

public class PhotoDataSourceFactory extends DataSource.Factory {

    private MutableLiveData<PageKeyedDataSource<Integer, Photo>> photoLiveDataSource = new MutableLiveData<>();

    @Override
    public DataSource create() {
        PhotoDataSource photoDataSource = new PhotoDataSource();
        photoLiveDataSource.postValue(photoDataSource);
        return photoDataSource;
    }

    public MutableLiveData<PageKeyedDataSource<Integer, Photo>> getPhotoLiveDataSource() {
        return photoLiveDataSource;
    }

}

File: app/src/main/java/com/example/unsplash/model/datasource/PhotoDataSourceFactory.kt
package com.example.unsplash.model.datasource

import androidx.lifecycle.MutableLiveData
import androidx.paging.DataSource
import androidx.paging.PageKeyedDataSource

import com.example.unsplash.model.models.Photo

class PhotoDataSourceFactory : DataSource.Factory<Int, Photo>() {

    private val photoLiveDataSource = MutableLiveData<PageKeyedDataSource<Int, Photo>>()

    override fun create(): DataSource<Int, Photo> {
        val photoDataSource = PhotoDataSource()
        photoLiveDataSource.postValue(photoDataSource)
        return photoDataSource
    }
}
